components/SpotlightFilters/SpotlightFilters.tsx

```diff
@@ -27,6 +29,15 @@ const SpotlightFilters: FunctionComponent<SpotlightFilterProps> = ({ defaultInde
         const { options: themeOptions, selected: themeSelected } = getThemeDefaults(selectedTheme, options);
         setSelected(themeSelected);
         setOptions(themeOptions);
+        const href = router.route;
+        const as = router.asPath + `?mapTopic=${themeSelected.theme?.name}`;
+        if (router.asPath == `spotlight/${router.query.slug}`) {
+          router.push(href, as, { shallow: true });
+        } else {
+          const asPath = router.asPath.split(/\?/)[0];
+          const as = asPath + `?mapTopic=${option.label}`;
+          router.push(href, as, { shallow: true });
+        }
```

Comment:
I have a feeling all this can be extracted into a reusable function, not sure. Give it some thought and see if you can identify any common bits you can use to build a smart function. It'd suck if you have this bulk of code in 3 places (topic, indicator & year).